code/game/objects/items/devices/PDA/PDA.dm

```diff
@@ -314,10 +316,9 @@ var/global/list/obj/item/device/pda/PDAs = list()
 				update_label()
 			if("Eject")//Ejects the cart, only done from hub.
 				if (!isnull(cartridge))
-					var/turf/T = loc
-					if(ismob(T))
-						T = T.loc
-					cartridge.loc = T
+					var/mob/M = loc

```

Comment:
also we already typecasted usr as a mob at the beginning of the proc, use that instead.